modules/edit-modes/src/types.js

```diff
@@ -63,3 +63,26 @@ export type PointerMoveEvent = {
   pointerDownMapCoords: ?Position,
   sourceEvent: any
 };
+
+export type ModeProps<TData> = {
```

Comment:
Add another field `viewport`?

Reply:
We can do that. How does this look?

```
export type Viewport = {
  longitude: number,
  latitude: number,
  zoom: number,
  bearing: number,
  pitch: number,
  width: number,
  height: number
};
```

Or do you need `project` and `unproject`?

Reply:
I need `project` / `unproject`.

Reply:
ok. let's do that part in a separate PR